fauna/client/client.py

```diff
@@ -375,6 +377,33 @@ def _query(
           schema_version=schema_version,
       )
 
+  @contextmanager
+  def stream(self, token: StreamToken) -> Iterator[Any]:
```

Comment:
what is the UX here if the client stream request fails?  There are 2 kinds of failure scenarios with the stream API currently:
1. the stream request outright fails, e.g. invalid auth token, invalid stream token
2. the stream fails during processing of events

in 1 you just get a failed response, in 2 you'll get an error event.  Wondering if we should differentiate or throw exceptions for the 1 scenario.

```
core-new (stream-log*) » curl -H "Authorization: Bearer BADAUTH" -H "Content-Type: application/json" -d "{\"token\":\"g9WD1YPGgmhUZXN0Q29sbIMAAGYqbnVsbCqByoaDAABmKm51bGwqgmNhbGyDAABmKm51bGwqgPT2gwAAZipudWxsKoMAAGYqbnVsbCqByoaDAABmKm51bGwqgmh0b1N0cmVhbYMAAGYqbnVsbCqA9PaDAABmKm51bGwqgwAAZipudWxsKoDBghpl5SRBGh8dFoA=\" }" http://localhost:8443/stream/1

{"error":{"code":"unauthorized","message":"Invalid token, unable to authenticate request"}}%
```

```
core-new (stream-log*) » curl -H "Authorization: Bearer secret" -H "Content-Type: application/json" -d "{\"token\":\"g9WD1YPVg8aCaFRlc3RDb2xsgwAAZipudWxsKoHKhoMAAGYqbnVsbCqCY2FsbIMAAGYqbnVsbCqA9PaDAABmKm51bGwqgwAAZipudWxsKoHKhoMAAGYqbnVsbCqCZCRtYXCDAABmKm51bGwqgsyEgYJlJHRoaXODAABmKm51bGwq9tWDxoJlJHRoaXODExNnKnF1ZXJ5KoLJg4MTFGcqcXVlcnkqgmVmaXJzdIMUGBlnKnF1ZXJ5KvTJg4MYGRgaZypxdWVyeSqCZnNlY29uZIMYGhggZypxdWVyeSr0gxMYIGcqcXVlcnkqgwAAZipudWxsKsaCYnYwgwAAZipudWxsKvT2gwAAZipudWxsKoMAAGYqbnVsbCqByoaDAABmKm51bGwqgmh0b1N0cmVhbYMAAGYqbnVsbCqA9PaDAABmKm51bGwqgwAAZipudWxsKoH2wYIaZeYtJRohrTdA\" }" http://localhost:8443/stream/1

{"type":"start","ts":1709583653565000,"stats":{"read_ops":8,"storage_bytes_read":144,"compute_ops":1,"processing_time_ms":10,"rate_limits_hit":[]}}
{"type":"error","code":"invalid_null_access","message":"Cannot access `second` on null.","stats":{"read_ops":1,"storage_bytes_read":49,"compute_ops":1,"processing_time_ms":20,"rate_limits_hit":[]}}
```

Reply:
1. if you pass a Query it will fail because it needs to perform a query before to return the StreamToken
2. you will receive error events from Fauna, in this case I'm closing the stream